ch.sbb.scion.rcp.workbench/src/ch/sbb/scion/rcp/workbench/popup/PopupCommand.java

```diff
@@ -0,0 +1,56 @@
+package ch.sbb.scion.rcp.workbench.popup;
+
+import java.util.UUID;
+
+public class PopupCommand {
+
+  public String popupId;
+
+  public PopupContext context;
+
+  public CloseStrategy closeStrategy;
+
+  public PopupCommand() {
+    this.popupId = UUID.randomUUID().toString();
+  }
+
+  public PopupCommand context(final PopupContext context) {
+    this.context = context;
+    return this;
+  }
+
+  public PopupCommand closeStrategy(final CloseStrategy closeStrategy) {
+    this.closeStrategy = closeStrategy;
+    return this;
+  }
+
+  public static class PopupContext {
+
+    public String viewId;
+
+    public PopupContext viewId(final String viewId) {
+      this.viewId = viewId;
+      return this;
+    }
+
+  }
+
+  public static class CloseStrategy {
```

Comment:
Do we really need both "PopupCloseStrategy" and "CloseStrategy"?

Reply:
I would like to keep them both. The CloseStrategy is used to deserialize the PopupCommand, which is somehow "internal API"; a PopupCommand can be issued from the web workbench popup service (TS), or from the rcp workbench popup service (Java). On the other hand, the PopupCloseStrategy is a purely rcp workbench popup capability implementation internal object. (The PopupContext and the PopupReferrer is another pair that was created based on this thought.) Does this make sense?

Reply:
Ok, then - without knowing for sure - do we have absolute minimal visibility of each of them? "public static" is equals to "global".
"CloseStrategy" should be private then.

Reply:
Actually, you are right - I simplified the implementation and removed the nested classes.